api/client/gro_client.py

```diff
@@ -212,7 +212,7 @@ def convert_unit(self, point, target_unit_id):
             new unit_id. Other properties are unchanged.
 
         """
-        if point.get('unit_id') is None or point.get('unit_id') == target_unit_id:
+        if point['value'] is None or point.get('unit_id') is None or point.get('unit_id') == target_unit_id:
```

Comment:
I think we should still swap the `point['unit_id']` value to the new unit id, if it passes the non-convertible checks.

A lot of people are going to be using this output in pandas data frames and pivoting on these columns, so if we leave it as the old unit_id, the Nones would get split off from the rest of the data if the non-None values got their ids updated but these didn't.

Reply:
Can do something like

```py
if point['value'] is not None:
    value_in_base_unit = (
        point['value'] * from_convert_factor.get('factor')
    ) + from_convert_factor.get('offset', 0)
    point['value'] = float(
        value_in_base_unit - to_convert_factor.get('offset', 0)
    ) / to_convert_factor.get('factor')
point['unit_id'] = target_unit_id
```

I still want those `if not *.get('factor'):` Exceptions to come before return though, for the same reason as the above ^. If one point's getting converted, they should all get converted

Reply:
Okay, I'll do something like that, agree that all of the points should end up in the same unit even if None